### ISSUE TITLE
[Feature Request]: Consider incorporating a status message into the Loading icon so that its resolution is surfaced to AT

### ISSUE THREAD

**Issue by mbgower** [carbon-design-system/carbon#12038]
### Summary

As described in the [accessibility review of the Loading component](https://ibm.box.com/s/q0l4vqxye2nihttqamy7h33111w3zmjd) (and to some degree in the Inline loading as well), although Carbon nicely surfaces the appearance of a Loading icon to Assistive Technologies (ATs), it doesn't surface the disappearance/resolution of the loading state.

This is important because to a sighted user, the disappearance of the Loading icon indicates that the loading state has ended.

If a component takes focus after loading completes, the system will notify users of the change of focus (and thus they can assume the loading has finished). However, where a page is loading or the interaction is such that there is no logical item to take focus, a user who cannot perceive the loading icon will never know when it has disappeared and the loading state has resolved. So in a situation where a load may take a number of seconds, they have to keep checking (in some way) if anything is operable. It's not optimal.

One thing Carbon could consider is adding a status message for ATs at the same time they cause the loading icon to disappear. It would need some user testing to ensure it improves the user experience, but would be a fairly simple change to make.

### Justification

Improves experience with users who cannot perceive the loading icon.

### Screenshots

See documentation linked to in Summary.

### Your team

IBM Accessibility

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
